projects/jupyter-server-ydoc/jupyter_server_ydoc/handlers.py

```diff
@@ -109,9 +109,26 @@ async def prepare(self):
                     # TransientRoom
                     # it is a transient document (e.g. awareness)
                     self.room = TransientRoom(self._room_id, self.log)
+            try:
+                await self._websocket_server.start_room(self.room)
+                self._websocket_server.add_room(self._room_id, self.room)
+            except Exception as e:
+                self.log.error(
+                    "Room %s failed to start on websocket server", self._room_id
+                )
+                # Clean room
+                self.room.stop()
+                del self.room
+                self.log.info("Room %s deleted", self._room_id)
+                self._emit(LogLevel.INFO, "clean", "Room deleted.")
 
-            await self._websocket_server.start_room(self.room)
-            self._websocket_server.add_room(self._room_id, self.room)
+                # Clean the file loader if there are not rooms using it
+                _, _, file_id = decode_file_path(self._room_id)
+                file = self._file_loaders[file_id]
+                self.log.info("Deleting file %s", file.path)
+                await self._file_loaders.remove(file_id)
+                self._emit(LogLevel.INFO, "clean", "Loader deleted.")
+                raise e
```

Comment:
Should the exception be raised, or just logged?

Reply:
it should be raised when task group is no active.